### PR TITLE
Annunciator

### PR DESCRIPTION
**Pull request by almatthew**
This PR supports [QIN-20](https://planckaero.atlassian.net/browse/QIN-20?atlOrigin=eyJpIjoiZjk0NjAyODI1NWNjNGNkOThhNTJiZTI3ZWE3ZGMyNGIiLCJwIjoiaiJ9) and [KA-67](https://planckaero.atlassian.net/browse/KA-67?atlOrigin=eyJpIjoiOTlkMTA3ZmNlOTVhNDYyNGE0Yjc5ZWEyMjY5YzA3Y2IiLCJwIjoiaiJ9) by introducing an "Annunciator Panel" to the top of the QGC flight window.  This PR does a few things:

1. Adds the optional `annunciatorPanel` object to the `FlightView` display in `FlightDisplayView.qml`
2. Adds support for consuming APM-specific `EKF_STATUS_REPORT` messages and converting them into `ESTIMATOR_STATUS` messages that QGC already understands.  Thankfully these messages are nearly identical.
3. Makes the `annunciatorPanel` optionally (default off) displayed with a General settings checkbox.

The annunciator panel itself displays "Traffic Light" indications of the health of various telemetry parameters, in this case:

1. EKF velocity health
2. EKF position (horiz & vert) health
3. EKF compass health
4. EKF terrain health
5. Vibration data
6. Lean angle

All parameters for the traffic light colors are hard-coded. EKF and vibe thresholds are based on values from MissionPlanner, which has a similar annunciator panel setup.  I pulled the lean angle parameters out of thin air.

Details on the requirements for this task are here: [QIN-23](https://planckaero.atlassian.net/browse/QIN-23?atlOrigin=eyJpIjoiZGExNzQxZmZlNWEyNDdhZDg2MWU2NTkzMzg0OTIwYzgiLCJwIjoiaiJ9)

### CODE REVIEW

**Comment by almatthew** [comment]
Here is a video of this in action: https://planckaerosystems.slack.com/files/U0WV8GD0B/F013RHG8LUW/annunciator_demo.mov

File: src/FlightDisplay/FlightDisplayView.qml [hunk 1]
```diff
@@ -728,6 +729,133 @@ Item {
...
     }
 
+    //-- Annunciator panel
+    Rectangle {
+        visible:                    _showAnnunciatorPanel
+        id:                         annunciatorPanel
+        anchors.horizontalCenter:   parent.horizontalCenter
+        anchors.topMargin:          _toolsMargin
+        anchors.top:                parent.top
+        z:                          _mapAndVideo.z + 4
```
Comment (ptran) [inline src/FlightDisplay/FlightDisplayView.qml]: The 4 seems arbitrary. Is there a way to get the count of all the items? Any concerns changes in the future could push the panel to be hidden?

File: src/FlightDisplay/FlightDisplayView.qml [hunk 2]
```diff
@@ -728,6 +729,133 @@ Item {
...
+            spacing:    ScreenTools.defaultFontPixelWidth
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.verticalCenter: parent.verticalCenter
+
+            Repeater {
+                id: annunciatorRepeater
+                property var ekfLowThresh: 0.5
+                property var ekfHighThresh: 0.8
+                property var vibeLowThresh: 30
+                property var vibeHighThresh: 50
```
Comment (ptran) [inline src/FlightDisplay/FlightDisplayView.qml]: Should these be facts vs hard coded?

**Review by ptran — COMMENTED** [review]
I think the code looks fine. Just left some comments.

File: src/FlightDisplay/FlightDisplayView.qml (hunk 2 above)
Comment (almatthew) [reply src/FlightDisplay/FlightDisplayView.qml]: @ptran not sure how to "hard code" these.  I made them `readonly property double`s...is that what you mean?

File: src/FlightDisplay/FlightDisplayView.qml (hunk 1 above)
Comment (almatthew) [reply src/FlightDisplay/FlightDisplayView.qml]: Good find, reduced this to be just above the map/video level.

File: src/FlightDisplay/FlightDisplayView.qml (hunk 2 above)
Comment (almatthew) [reply src/FlightDisplay/FlightDisplayView.qml]: Oh, if you're asking why I didn't put them inline in the code: These might become parameters in the future that are modifiable by the user.  TBD.

File: src/FlightDisplay/FlightDisplayView.qml [hunk 3]
```diff
@@ -728,6 +729,140 @@ Item {
...
+                    return getColor(
+                                getLargestOf(
+                                    activeVehicle.vibration.xAxis.value,
+                                    activeVehicle.vibration.yAxis.value,
+                                    activeVehicle.vibration.zAxis.value),
+                                vibeLowThresh, vibeHighThresh)
+                }
+
+                function getTiltColor() {
+                    return getColor(
```
Comment (AubreyC) [inline src/FlightDisplay/FlightDisplayView.qml:811]: Does it handle negative value ? (can tilt be negative?)

Comment (almatthew) [reply src/FlightDisplay/FlightDisplayView.qml:811]: `getAbsLargestOf` uses the absolute value of each.  I've renamed the method for clarity.

**Comment by almatthew** [comment]
I made a couple changes to this PR at the customer's request:
1. Parameterized the tilt and vibe max values: Users can set them in the General settings
2. Made the colors scale linearly green->yellow->red up to the maximum value
I also cleaned the code a bit.

**Comment by AubreyC** [comment]
@almatthew Getting fancy with the continuous color changes, looks good

**Review by AubreyC — APPROVED** [review]

**Comment by almatthew** [comment]
Tested successfully in the cage on Windows and Android.

**Pull request merged**